platform/admin/core/admin-core-impl/src/test/java/org/codice/ddf/admin/core/impl/SystemPropertiesAdminTest.java

```diff
@@ -208,6 +183,31 @@ public void testWriteSystemProperties() throws Exception {
     }
   }
 
+  @Test
+  public void testWritePlaceholderProperties() throws Exception {
+    SystemPropertiesAdmin spa = new SystemPropertiesAdmin(mockGuestClaimsHandlerExt);
+    Map<String, String> map = new HashMap<>();
+    map.put(SystemBaseUrl.INTERNAL_HOST, "localhost");
+    map.put(SystemBaseUrl.INTERNAL_HTTP_PORT, "5678");
+    map.put(SystemBaseUrl.INTERNAL_HTTPS_PORT, "1234");
+    map.put(SystemBaseUrl.EXTERNAL_HOST, "localhost");
+    map.put(SystemBaseUrl.EXTERNAL_HTTP_PORT, "5678");
+    map.put(SystemBaseUrl.EXTERNAL_HTTPS_PORT, "1234");
+
+    spa.writeSystemProperties(map);
+    List<SystemPropertyDetails> details = spa.readSystemProperties();
+    assertThat(getDetailsValue(details, SystemBaseUrl.EXTERNAL_HOST), equalTo("localhost"));
+    assertThat(getDetailsValue(details, SystemBaseUrl.EXTERNAL_HTTP_PORT), equalTo("5678"));
+    assertThat(getDetailsValue(details, SystemBaseUrl.EXTERNAL_HTTPS_PORT), equalTo("1234"));
+    assertThat(getDetailsValue(details, SystemBaseUrl.INTERNAL_HOST), equalTo("localhost"));
+    assertThat(getDetailsValue(details, SystemBaseUrl.INTERNAL_HTTP_PORT), equalTo("5678"));
+    assertThat(getDetailsValue(details, SystemBaseUrl.INTERNAL_HTTPS_PORT), equalTo("1234"));
+    assertThat(getDetailsValue(details, SystemInfo.ORGANIZATION), equalTo("org"));
+    assertThat(getDetailsValue(details, SystemInfo.SITE_CONTACT), equalTo("contact"));
+    assertThat(getDetailsValue(details, SystemInfo.SITE_NAME), equalTo("site"));
+    assertThat(getDetailsValue(details, SystemInfo.VERSION), equalTo("version"));
+  }
+
   private String getDetailsValue(List<SystemPropertyDetails> props, String key) {
```

Comment:
✏️ I know this wasn't part of your changes, but this could be replaced with a custom Matcher
```java
assertThat(
    spa.readSystemProperties(),
    contains(
        hasKeyAndValue(SystemBaseUrl.EXTERNAL_HOST, "localhost"),
        hasKeyAndValue(SystemBaseUrl.EXTERNAL_HTTP_PORT, "5678"),
        hasKeyAndValue(SystemBaseUrl.EXTERNAL_HTTPS_PORT, "1234"),
        hasKeyAndValue(SystemBaseUrl.INTERNAL_HOST, "localhost"),
        hasKeyAndValue(SystemBaseUrl.INTERNAL_HTTP_PORT, "5678"),
        hasKeyAndValue(SystemBaseUrl.INTERNAL_HTTPS_PORT, "1234"),
        hasKeyAndValue(SystemInfo.ORGANIZATION, "Codice Foundation"),
        hasKeyAndValue(SystemInfo.SITE_CONTACT, ""),
        hasKeyAndValue(SystemInfo.SITE_NAME, "ddf.distribution"),
        hasKeyAndValue(SystemInfo.VERSION, "2.14.0-SNAPSHOT")));
```
```java
private static Matcher<SystemPropertyDetails> hasKeyAndValue(
    final String key, final String value) {
  return new TypeSafeMatcher<SystemPropertyDetails>() {
    @Override
    public void describeTo(Description description) {
      description.appendText(String.format("key=%s and value=%s", key, value));
    }

    @Override
    protected boolean matchesSafely(SystemPropertyDetails systemPropertyDetails) {
      return key.equals(systemPropertyDetails.getKey())
          && value.equals(systemPropertyDetails.getValue());
    }
  };
}
```